Fix double-escaping of newlines

Bryght-Labs-Richard opened the pull request:
I'm not sure this is the right approach. It looks like lib/parser.js has a mechanism to un-escape input files, and lib/compiler_template.js has a mechanism to escape output files, and that seemed to make sense, so I was expecting both to run, but it seems `esc()` was running on the raw text, which was then double-escaped.

This change removes `esc()`.

Fixes https://github.com/lvgl/lv_i18n/issues/66

Bryght-Labs-Richard committed:
fix double-escaping of newlines

Fixes https://github.com/lvgl/lv_i18n/issues/66

teaalltr commented:
Hi @bryghtlabs-richard it works for me, could you merge the PR? Thank you!

paaspaas00 commented:
> Hi @bryghtlabs-richard it works for me, could you merge the PR? Thank you!

@bryghtlabs-richard hold on, I'm testing with apostrophe escaping ( \' ), will be able to tell if it works properly in a day or so

bryghtlabs-richard commented:
Hi all, I'm not a maintainer of lv_i18n - we'll need their review and help, and with anything it's worth taking the time to get it right. Also, I'm not a JS developer, please take this patch with a grain of salt, and I'm not sure this is the right way to handle this - the other option will be to ensure the keys/values get de-escaped on parsing, then re-escaped on output.

kisvegabor commented:
cc @puzrin 

So we assume that the characters are correctly escaped in the input strings already, so we can use them as they are, right?

bryghtlabs-richard commented:
It seems to work that way. There are some tests for C unescaping, I'll try to get those running when I look at this again. @rlidwka , any thoughts?

vortex-surfer commented:
Hi @bryghtlabs-richard, I rebased your commit on master. It seems to work for me but I also am not a JS developer and cannot comment on any possible corner cases.
[0001-fix-double-escaping-of-newlines.patch](https://github.com/user-attachments/files/18375764/0001-fix-double-escaping-of-newlines.patch)

raoulh commented:
I came across this exact bug, why isn't this PR merged?

bryghtlabs-richard commented:
I haven't had time to finish testing each escape sequence. Feel free to pick this up.

kisvegabor commented:
Can someone pick up this PR and test with the [proposed patch](https://github.com/lvgl/lv_i18n/pull/68#issuecomment-2582423156)?

bryghtlabs-richard commented:
I looked at this again. Here's why I don't understand it: if JavaScript has a line containing a newline character, it should absolutely be escaped. Is the YAML parser not converting '\n' to a newline character? Probably not - I don't think that means anything to Yaml.

I came to think that I should make my YML files contain a line break using the `?:` syntax, but those lines get joined with spaces into a key, so that doesn't work.

Maybe the answer is to replace translation keys with newlines with an intermediate text key without newlines, then translate those to all languages(including the default lang)?

Bryght-Labs-Richard committed:
Fix cli tests on Windows (#87)

Call the node executable directly so we don't have to rely exec knowing how to do it.

Bryght-Labs-Richard committed:
fix: join/path separator usage

Pass in the arguments as they come, and let join handle them.

AndreCostaaa commented:
Note the conflicts

Bryght-Labs-Richard committed:
fix: temporarily downgrade glob so we can run tests again

lv_i18n is relying on some unintentional behaviour within glob <8 on Windows.
Until we can address that, back down to the latest glob 7.y.z

bryghtlabs-richard commented:
Oops, please wait, working on `--optimize` mode. EDIT: no issues.

bryghtlabs-richard commented:
All, I'd appreciate if others could try this out as well. The thing I worry about is any other escape sequences that might be used.

@AndreCostaaa , this is built atop the [fixCliTestOnWindows ](https://github.com/lvgl/lv_i18n/pull/88) branch, which could be merged separately.
